packages/governance/src/binaryVoteCounter.js

```diff
@@ -106,7 +107,7 @@ const makeBinaryVoteCounter = (questionSpec, threshold, instance) => {
       if (choice < 0) {
         spoiled += shares;
       } else {
-        tally[choice] += shares;
+        tally[+choice] += shares;
```

Comment:
What does a unary plus do as an array index? Is it coercing `choice` to a number?

Reply:
Yes, I think so.

I have only a vague understanding of why, but...

> Jessie omits mutation through computed property names. Jessie has syntax for mutating only number-named properties, which include integers, floating point, NaN, Infinity, and -Infinity.
 -- https://github.com/endojs/Jessie

Reply:
We want to make it easier to statically reason more reliably about Jessie programs, while avoiding undue syntactic burden. A typing practice common outside JS and approximated by TS is that records are heterogeneously typed -- each named field is separately typed, and arrays are uniformly typed. By JS fights us here, by making array indexes just be normal property names, and generalizing the `[]` syntax to apply to all property names. Computed indexing destroys simple static reasoning about heterogeneously typed record fields.

The more correct syntax for Jessie to insist on to ensure computed property names are array indexes would be `obj[index|0]` as done in asm.js. Perhaps we should do that instead. But we opted for a looser compromise to get a slightly less unpleasant syntax. Our looser compromise depends on normal records not having properties named for floating point values or `NaN` or `Infinity` or `-Infinity`.

(Btw, I think the `obj[+index]` syntactic compromise is originally due to @mikesamuel . Thanks!)

Reply:
The 
```js
E(paramMgr)[`update${paramName}`]
```
below is a great example of something that is prohibited by Jessie, but is in this case a good idea. For this one, I think the correct approach is an appropriate eslint-ignore-like comment or ts-ignore-like comment, though I don't yet know how to do those for Jessie. It is the right approach because it is *unusual* for code like this to be a good idea, and its costs to static reasoning are real and significant. As with ts-ignore comments, the Jessie ignore comment should be accompanied by rationale.

Reply:
For the sort of formal methods I hope to use, I don't expect them to cope well with `ignore` comments.
I'm inclined to extract the computed property lookup stuff to a function in a separate file, which we would reason about separately.

Reply:
Yes. Along the lines of your `fromEntries` suggestion at https://github.com/Agoric/agoric-sdk/pull/3876#issuecomment-925089605 , for each of these accesses, there is already adequate reflective API. For property reading by itself, for example, there is

```js
Reflect.get(E(paramMgr), `update${paramName}`)
```

Reply:
I had the opposite opinion earlier, but now I don't see many gains from `E(paramMgr)[`update${paramName}`]`. What makes this a good idea?

Reply:
The [earlier answer](https://github.com/Agoric/agoric-sdk/pull/3448#discussion_r673345149) was:

>  each updateFoo() message is a separable capability.

I mentioned "my preference for static typing", which argues for your `E(paramMgr).update(paramName, proposedValue)` suggestion; it looks like Jessie suggests a similar preference.

I suppose the separate capability could always be expressed using currying: `proposedValue => E(paramMgr).update("knob1", proposedValue)`.

I can see it both ways.